components/ciphertool/src/main/java/org/wso2/ciphertool/cipher/SymmetricCipher.java

```diff
@@ -0,0 +1,173 @@
+    private static final String GET_KEY_ERROR_MESSAGE = "Error retrieving key associated with alias : ";
+    private static final String CIPHER_INIT_ERROR_MESSAGE = "Error initializing Cipher.";
+    private static final String INVALID_SECRET_ERROR_MESSAGE = "The provided secret key is invalid.";
+
+    public SymmetricCipher(KeyStore keyStore, String keyAlias) {
+
+        String cipherTransformation = System.getProperty(Constants.CIPHER_TRANSFORMATION_SYSTEM_PROPERTY);
+        this.algorithm = StringUtils.isNotBlank(cipherTransformation)
+                ? cipherTransformation : Constants.AES_GCM_NO_PADDING;
+        String password = KeyStoreUtil.getKeystorePassword();
+        try {
+            this.secretKey = keyStore.getKey(keyAlias, password.toCharArray());
+            if (this.secretKey == null) {
+                throw new KeyStoreException(GET_KEY_ERROR_MESSAGE + keyAlias);
+            }
+            this.cipher = Cipher.getInstance(this.algorithm);
+        } catch (NoSuchAlgorithmException | NoSuchPaddingException e) {
+            throw new CipherToolException(CIPHER_INIT_ERROR_MESSAGE, e);
+        } catch (KeyStoreException | UnrecoverableKeyException e) {
+            throw new CipherToolException(GET_KEY_ERROR_MESSAGE + keyAlias, e);
+        }
+    }
+
+    public SymmetricCipher(KeyStore keyStore) {
+
+        this(keyStore, System.getProperty(Constants.KEY_ALIAS_PROPERTY));
+    }
+
+    /**
+     * Encrypt plain text using encryption.
+     *
+     * @param plainText Plain text password.
+     * @return Encrypted password.
+     */
+    @Override
+    public String doEncryption(String plainText) {
+
+        try {
+            if (Constants.AES_GCM_NO_PADDING.equals(this.algorithm)) {
+                byte[] iv = getInitializationVector();
+                cipher.init(Cipher.ENCRYPT_MODE, this.secretKey, new GCMParameterSpec(GCM_TAG_LENGTH, iv));
+                String cipherText = Utils.doEncryption(cipher, plainText);
+                return createSelfContainedCiphertextWithGCMMode(cipherText, iv);
+            } else {
+                cipher.init(Cipher.ENCRYPT_MODE, this.secretKey);
+                return Utils.doEncryption(cipher, plainText);
+            }
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new CipherToolException(CIPHER_INIT_ERROR_MESSAGE, e);
+        }  catch (InvalidKeyException e) {
+            throw new CipherToolException(INVALID_SECRET_ERROR_MESSAGE, e);
+        }
+    }
+
+    /**
+     * Decrypt encrypted text using encryption.
+     *
+     * @param cipherText Encrypted password.
+     * @return Plain text password.
+     */
+    @Override
+    public String doDecryption(String cipherText) {
+
+        try {
+            byte[] encryptedText;
+            if (Constants.AES_GCM_NO_PADDING.equals(this.algorithm)) {
+                JsonObject jsonObject = getJsonObject(cipherText);
+                encryptedText = getValueFromJson(jsonObject, "cipherText");
```

Comment:
Introduce constant for "cipherText" and "iv" since there are usages in multiple methods.